src/infra/core/src/repos/metadata_chain_impl.rs

```diff
@@ -415,11 +415,13 @@ where
 
     async fn accept_append_validators<'a>(
         &'a self,
-        decisions: &'a mut [Decision],
+        decisions: &'a mut [MetadataVisitorDecision],
         visitors: &mut [&mut dyn MetadataChainVisitor<Error = AppendError>],
         prev_append_block_hash: Option<&'a Multihash>,
     ) -> Result<(), AppendError> {
-        let have_already_stopped = decisions.iter().all(|decision| *decision == Decision::Stop);
+        let have_already_stopped = decisions
+            .iter()
+            .all(|decision| *decision == MetadataVisitorDecision::Stop);
```

Comment:
> Minor: maybe holding a counter of active visitors among with `decisions` slice could reduce this comparison to 1 operation, on the other hand we would have to detect changes in `decisions` within the traversal code. (c) https://github.com/kamu-data/kamu-cli/pull/523#discussion_r1517474271

Reply:
In this case, the validation of the earlier decisions (of Validators) is one-time and tentative -- I don't think we're going to get a significant advantage this time